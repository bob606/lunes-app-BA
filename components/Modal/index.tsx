import {
  React,
  Text,
  TouchableOpacity,
  View,
  styles,
  Modal,
  IConfirmationModalProps,
  CloseIcon,
  Button,
  BUTTONS_THEME,
  AsyncStorage,
} from './imports';

const ConfirmationModal = ({
  visible,
  setIsModalVisible,
  navigation,
  extraParams,
}: IConfirmationModalProps) => {
  const closeModal = () => setIsModalVisible(false);

  const goBack = () => {
    setIsModalVisible(false);
    AsyncStorage.removeItem('session');
    navigation.navigate('Exercises', {extraParams});
  };

  return (
    <Modal
      testID="modal"
      visible={visible}
      transparent
      animationType="fade"
      style={styles.container}>
      <View style={[styles.container, styles.overlay]}>
        <View style={styles.modal}>
          <TouchableOpacity style={styles.closeIcon} onPress={closeModal}>
            <CloseIcon />
          </TouchableOpacity>
          <Text style={styles.message}>
            Do you really want to end this session?
          </Text>
          <Button onPress={closeModal} theme={BUTTONS_THEME.dark}>
            <Text style={styles.lightLabel}>continue</Text>
          </Button>

          <Button onPress={goBack} theme={BUTTONS_THEME.light}>
            <Text style={styles.darkLabel}>end</Text>
          </Button>
        </View>
      </View>
    </Modal>
  );
};

export default ConfirmationModal;
