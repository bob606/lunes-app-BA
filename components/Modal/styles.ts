import {StyleSheet} from 'react-native';
import {COLORS} from '../../constants/colors';
import {widthPercentageToDP as wp} from 'react-native-responsive-screen';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
  },
  overlay: {
    marginTop: 0,
    backgroundColor: COLORS.lunesOverlay,
  },
  modal: {
    backgroundColor: COLORS.white,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 260,
    width: wp('85%'),
    borderRadius: 4,
    position: 'relative',
  },
  closeIcon: {
    position: 'absolute',
    top: 8,
    right: 8,
    width: 24,
    height: 24,
  },
  message: {
    textAlign: 'center',
    fontSize: 20,
    color: COLORS.lunesGreyDark,
    fontFamily: 'SourceSansPro-SemiBold',
    width: wp('60%'),
    marginBottom: 31,
  },
  continueLabel: {
    color: COLORS.lunesWhite,
    fontSize: 14,
    fontWeight: '600',
    textAlign: 'center',
    textTransform: 'uppercase',
    fontFamily: 'SourceSansPro-SemiBold',
    letterSpacing: 0.4,
  },
  endLabel: {
    color: COLORS.lunesBlack,
    fontSize: 14,
    fontWeight: '600',
    textAlign: 'center',
    textTransform: 'uppercase',
    fontFamily: 'SourceSansPro-SemiBold',
    letterSpacing: 0.4,
  },
});
