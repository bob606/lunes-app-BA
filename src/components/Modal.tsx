import React from 'react'
import { View, Text, TouchableOpacity, Modal, StyleSheet } from 'react-native'
import { CloseIcon } from '../../assets/images'
import Button from './Button'
import { BUTTONS_THEME } from '../constants/data'
import { COLORS } from '../constants/colors'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
import AsyncStorage from '../services/AsyncStorage'
import labels from '../constants/labels.json'
import { StackNavigationProp } from '@react-navigation/stack'
import { RoutesParamsType } from '../navigation/NavigationTypes'
import { RouteProp } from '@react-navigation/native'
import styled from 'styled-components/native'

const ModalContainer = styled.Modal`
    flex: 1;
    justify-content: center;
    align-items: center;
`;
const Container = styled.View`
    flex: 1;
    justify-content: center;
    align-items: center;
    margin-top: 0;
    background-color: ${COLORS.lunesOverlay};
`;
const ModalStyle = styled.View`
    background-color: ${COLORS.white};
    align-items: center;
    justify-content: center;
    width: ${wp('85%')};
    border-radius: 4;
    position: relative;
    padding-top: 31;
    padding-bottom: 31;
`;
const CloseIconStyle = styled.TouchableOpacity`
    position: absolute;
    top: 8;
    right: 8;
    width: 24;
    height: 24;
`;
const Message = styled.Text`
    text-align: center;
    font-size: ${wp('5%')};
    color: ${COLORS.lunesGreyDark};
    font-family: 'SourceSansPro-SemiBold';
    width: ${wp('60%')};
    margin-bottom: 31;
    padding-top: 31;
`;
const LightLabel = styled.Text`
    color: ${COLORS.lunesWhite};
    font-size: ${wp('4%')};
    font-weight: 600;
    text-align: center;
    text-transform: uppercase;
    font-family: 'SourceSansPro-SemiBold';
    letter-spacing: 0.4;
`;
const DarkLabel = styled.Text`
    color: ${COLORS.lunesBlack};
    font-size: ${wp('4%')};
    font-weight: 600;
    text-align: center;
    text-transform: uppercase;
    font-family: 'SourceSansPro-SemiBold';
    letter-spacing: 0.4;
`;

export interface ConfirmationModalPropsType {
  visible: boolean
  setIsModalVisible: Function
  navigation: StackNavigationProp<RoutesParamsType, 'WordChoiceExercise' | 'ArticleChoiceExercise' | 'WriteExercise'>
  route: RouteProp<RoutesParamsType, 'WordChoiceExercise' | 'ArticleChoiceExercise' | 'WriteExercise'>
}

const ConfirmationModal = ({
  navigation,
  route,
  visible,
  setIsModalVisible
}: ConfirmationModalPropsType): JSX.Element => {
  const closeModal = (): void => setIsModalVisible(false)
  const goBack = (): void => {
    setIsModalVisible(false)
    AsyncStorage.clearSession().catch(e => console.error(e))
    const { disciplineID, disciplineTitle, disciplineIcon, trainingSetId, trainingSet } = route.params.extraParams
    const extraParams = {
      extraParams: {
        disciplineID: disciplineID,
        disciplineTitle: disciplineTitle,
        disciplineIcon: disciplineIcon,
        trainingSetId: trainingSetId,
        trainingSet: trainingSet
      }
    }
    navigation.navigate('Exercises', extraParams)
  }
  return (
    <ModalContainer testID='modal' visible={visible} transparent animationType='fade' >
      <Container>
        <ModalStyle>
          <CloseIconStyle onPress={closeModal}>
            <CloseIcon />
          </CloseIconStyle>
          <Message>{labels.exercises.cancelModal.cancelAsk}</Message>
          <Button onPress={closeModal} theme={BUTTONS_THEME.dark}>
            <LightLabel>{labels.exercises.cancelModal.continue}</LightLabel>
          </Button>
          <Button onPress={goBack} theme={BUTTONS_THEME.light}>
            <DarkLabel>{labels.exercises.cancelModal.cancel}</DarkLabel>
          </Button>
        </ModalStyle>
      </Container>
    </ModalContainer>
  )
}
export default ConfirmationModal