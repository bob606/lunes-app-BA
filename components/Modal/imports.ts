import React from 'react';
import {View, Text, TouchableOpacity, Modal} from 'react-native';
import {styles} from './styles';
import {IConfirmationModalProps} from '../../interfaces/exercise';
import {CloseIcon} from '../../assets/images/imports';
import Button from '../Button';
import {BUTTONS_THEME} from '../Button/imports';
import AsyncStorage from '@react-native-community/async-storage';

export {
  React,
  View,
  Text,
  TouchableOpacity,
  Modal,
  styles,
  CloseIcon,
  Button,
  BUTTONS_THEME,
  AsyncStorage,
};

export type {IConfirmationModalProps};
