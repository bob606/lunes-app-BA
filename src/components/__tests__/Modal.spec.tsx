import { View } from 'react-native'
import React from 'react'
import Modal, { ConfirmationModalPropsType } from '../Modal'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('Components', () => {
  describe('Modal ', () => {
    const defaultModalProps: ConfirmationModalPropsType = {
      navigation: '',
      setIsModalVisible: () => {},
      visible: false
    }

    it('renders correctly across screens', () => {
      const component = shallow(
        <View>
          <Modal {...defaultModalProps} />
        </View>
      )
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should have visible property passed to it as default', () => {
      const modalProps: ConfirmationModalPropsType = {
        ...defaultModalProps
      }

      const component = shallow(
        <View>
          <Modal {...modalProps} />
        </View>
      )
      expect(component.children().props().visible).toBe(false)
    })

    it('should have visible property passed to it', () => {
      const modalProps: ConfirmationModalPropsType = {
        ...defaultModalProps,
        visible: true
      }

      const component = shallow(
        <View>
          <Modal {...modalProps} />
        </View>
      )
      expect(component.children().props().visible).toBe(true)
    })
  })
})
