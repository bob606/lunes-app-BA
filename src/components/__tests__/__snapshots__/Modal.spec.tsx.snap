// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Modal  renders correctly across screens 1`] = `
<View>
  <ConfirmationModal
    navigation={
      Object {
        "addListener": [MockFunction],
        "canGoBack": [MockFunction],
        "dangerouslyGetParent": [MockFunction],
        "dangerouslyGetState": [MockFunction],
        "dispatch": [MockFunction],
        "goBack": [MockFunction],
        "isFocused": [MockFunction],
        "navigate": [MockFunction],
        "pop": [MockFunction],
        "popToTop": [MockFunction],
        "push": [MockFunction],
        "removeListener": [MockFunction],
        "replace": [MockFunction],
        "reset": [MockFunction],
        "setOptions": [MockFunction],
        "setParams": [MockFunction],
      }
    }
    route={
      Object {
        "key": "",
        "name": "WordChoiceExercise",
        "params": Object {
          "extraParams": Object {
            "disciplineID": 0,
            "disciplineIcon": "Icon",
            "disciplineTitle": "Title",
            "exercise": 1,
            "exerciseDescription": "Description",
            "level": [MockFunction],
            "trainingSet": "Set",
            "trainingSetId": 0,
          },
        },
      }
    }
    setIsModalVisible={[Function]}
    visible={false}
  />
</View>
`;
